crates/oxc_transformer/src/plugins/module_runner_transform.rs

```diff
@@ -0,0 +1,2055 @@
+__vite_ssr_import_0__.default.resolve('server.js');",
+        );
+    }
+
+    #[test]
+    fn import_meta() {
+        test_same("console.log(import.meta.url)", "console.log(__vite_ssr_import_meta__.url);");
+    }
+
+    #[test]
+    fn dynamic_import() {
+        test_same(
+            "export const i = () => import('./foo')",
+            "const i = () => __vite_ssr_dynamic_import__('./foo');
+Object.defineProperty(__vite_ssr_exports__, 'i', {
+  enumerable: true,
+  configurable: true,
+  get() {
+    return i;
+  }
+});",
+        );
+    }
+
+    #[test]
+    fn do_not_rewrite_method_definition() {
+        test_same(
+            "import { fn } from 'vue';class A { fn() { fn() } }",
+            "const __vite_ssr_import_0__ = await __vite_ssr_import__('vue', { importedNames: ['fn'] });
+class A {
+  fn() {
+    (0, __vite_ssr_import_0__.fn)();
+  }
+}",
+        );
+    }
+
+    #[test]
+    fn do_not_rewrite_when_variable_in_scope() {
+        test_same(
+            "import { fn } from 'vue';function A(){ const fn = () => {}; return { fn }; }",
+            "const __vite_ssr_import_0__ = await __vite_ssr_import__('vue', { importedNames: ['fn'] });
+function A() {
+  const fn = () => {};
+  return { fn };
+}",
+        );
+    }
+
+    #[test]
+    fn do_not_rewrite_destructuring_object() {
+        test_same(
+            "import { fn } from 'vue';function A(){ let {fn, test} = {fn: 'foo', test: 'bar'}; return { fn }; }",
+            "const __vite_ssr_import_0__ = await __vite_ssr_import__('vue', { importedNames: ['fn'] });
+function A() {
+  let { fn, test } = {
+    fn: 'foo',
+    test: 'bar'
+  };
+  return { fn };
+}",
+        );
+    }
+
+    #[test]
+    fn do_not_rewrite_destructuring_array() {
+        test_same(
+            "import { fn } from 'vue';function A(){ let [fn, test] = ['foo', 'bar']; return { fn }; }",
+            "const __vite_ssr_import_0__ = await __vite_ssr_import__('vue', { importedNames: ['fn'] });
+function A() {
+  let [fn, test] = ['foo', 'bar'];
+  return { fn };
+}",
+        );
+    }
+
+    #[test]
+    fn do_not_rewrite_catch_clause() {
+        test_same(
+            "import {error} from './dependency';try {} catch(error) {}",
+            "const __vite_ssr_import_0__ = await __vite_ssr_import__('./dependency', { importedNames: ['error'] });
+try {} catch (error) {}",
+        );
+    }
+
+    #[test]
+    fn should_declare_imported_super_class() {
+        test_same(
+            "import { Foo } from './dependency'; class A extends Foo {}",
+            "const __vite_ssr_import_0__ = await __vite_ssr_import__('./dependency', { importedNames: ['Foo'] });
+class A extends __vite_ssr_import_0__.Foo {}",
+        );
+    }
+
+    #[test]
+    fn should_declare_exported_super_class() {
+        test_same(
+            "import { Foo } from './dependency'; export default class A extends Foo {} export class B extends Foo {}",
```

Comment:
```suggestion
            "import { Foo } from './dependency'; export default class A extends Foo {}; export class B extends Foo {}",
```
Isn't `;` or `\n` needed here? My brain interpreter fails to parse this code 🤔

Reply:
Fixed in https://github.com/oxc-project/oxc/pull/9371/commits/67d2504c1579aa70bf29f5c085a778bddfced2a7